src/pages/workspaces/workspace/Data.js

```diff
@@ -732,14 +759,44 @@ const WorkspaceData = _.flow(
             isFeaturePreviewEnabled('workspace-data-service') && h(DataTypeSection, {
               title: 'WDS'
             }, [
-              div({
-                style: {
-                  display: 'flex', flexDirection: 'column', alignItems: 'flex-start',
-                  padding: '0.5rem 1.5rem', borderBottom: `1px solid ${colors.dark(0.2)}`,
-                  backgroundColor: 'white'
-                }
-              }, ['Coming soon.']),
-              div({}, '')
+              [
+                wdsSchemaError && h(NoDataPlaceholder, {
+                  message: 'WDS is unavailable.'
+                }),
+                wdsSchema && _.map(typeDef => {
+                  return div({ key: typeDef.name, role: 'listitem' }, [
+                    h(DataTypeButton, {
+                      key: typeDef.name,
+                      selected: selectedData?.type === workspaceDataTypes.wds && selectedData.entityType === typeDef.name,
+                      entityName: typeDef.name,
+                      entityCount: typeDef.count,
+                      filteredCount: typeDef.count,
+                      activeCrossTableTextFilter: false,
+                      crossTableSearchInProgress: false,
+                      onClick: () => {
+                        setSelectedData({ type: workspaceDataTypes.wds, entityType: typeDef.name })
+                        forceRefresh()
+                      },
+                      after: h(DataTableActions, {
+                        dataProvider: wdsDataTableProvider,
+                        tableName: typeDef.name,
+                        rowCount: typeDef.count,
+                        entityMetadata,
+                        workspace,
+                        onRenameTable: undefined,
+                        onDeleteTable: tableName => {
+                          setSelectedData(undefined)
+                          setWdsSchema(_.remove(typeDef => typeDef.name === tableName, wdsSchema))
+                          forceRefresh() // TODO: this probably isn't correct
```

Comment:
Why isn't this correct?

Reply:
ah good catch. It's related to https://github.com/DataBiosphere/terra-ui/pull/3429/files#diff-8c26354dc5cab7f1d3c38c6ea2ae1a195a5bcfff6668b119a7d7de3a7ef77ca5R21; I have updated the code comment to reflect that.